### ISSUE TITLE
Index not showing paged URLs for taxonomies or users.

### ISSUE THREAD

**Issue by joemcgill** [GoogleChromeLabs/wp-sitemaps#78]
**Describe the bug**
Following #72, the sitemap index now automatically shows all available sitemap pages for post types. However, it's not showing available sitemap pages for taxonomies, even when they're available.

**To Reproduce**
1. Generate > 2000 categories and add each of them to a post (I know 😫)
2. Visit the sitemap index page.
3. See that a sitemap for `sitemap-taxonomies-category-1.xml` is listed but `sitemap-taxonomies-category-2.xml` is not.
4. Manually go to `sitemap-taxonomies-category-2.xml` in your browser to see that the page actually does exist and includes URLs.

**Expected behavior**
All available taxonomy type sitemaps should exist in the sitemap index.

**Additional context**
This is hard to test right now because you have to have over 2000 terms with published posts available, so this would be easier to work on after #74 is completed.

**Comment by joemcgill** [comment]
Now that I've gotten #74 working via #82 (in review), I can confirm that this is also a problem for users. I think it makes sense to handle both cases in this ticket rather than opening a new ticket since the solution for both is probably similar, if not identical.

Once #82 lands, an easier way to test this is to filter `core_sitemaps_max_urls` (see the example in #82) so you don't have to generate so many objects in order to observe this bug.
